src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpContextOfT.cs

```diff
@@ -90,6 +92,7 @@ public IISHttpContextOfT(PipeFactory pipeFactory, IHttpApplication<TContext> app
                     await _readingTask;
                 }
             }
+            return failed ? NativeMethods.REQUEST_NOTIFICATION_STATUS.RQ_NOTIFICATION_FINISH_REQUEST : NativeMethods.REQUEST_NOTIFICATION_STATUS.RQ_NOTIFICATION_CONTINUE;
```

Comment:
Why do you finish the request if this failed??

Reply:
Do we want other modules to execute if there was an application failure? Seems intuitive to finish the request if there is a failure.

Reply:
> Modules that implement request handling should return RQ_NOTIFICATION_FINISH_REQUEST when an error occurs.

https://msdn.microsoft.com/en-us/library/ms690802(v=vs.90).aspx

Seems like we should be using FINISH_REQUEST according to docs. I'm not sure how error handling modules in IIS plug in though. In theory, we still want those to run.

Reply:
I see. I'd prefer it if we handled this all in a single place. Right now it's split across IISContextOfT and IISServer (to handle the synchronous case).

Reply:
We can make the Task return a bool for success or failure? Then all of the RequestNotification returns would be in IISServer.

Reply:
Would it be better to rethrow the exception instead of returning Task<T>